main.tf

```diff
@@ -46,19 +49,23 @@ resource "azurerm_kubernetes_cluster" "main" {
 
 
 resource "azurerm_log_analytics_workspace" "main" {
+  count               = var.enable_log_analytics_workspace ? 1 : 0
   name                = "${var.prefix}-workspace"
   location            = data.azurerm_resource_group.main.location
   resource_group_name = var.resource_group_name
   sku                 = var.log_analytics_workspace_sku
   retention_in_days   = var.log_retention_in_days
+
+  tags = var.tags
 }
 
 resource "azurerm_log_analytics_solution" "main" {
+  count                 = var.enable_log_analytics_workspace ? 1 : 0
   solution_name         = "ContainerInsights"
   location              = data.azurerm_resource_group.main.location
   resource_group_name   = var.resource_group_name
-  workspace_resource_id = azurerm_log_analytics_workspace.main.id
-  workspace_name        = azurerm_log_analytics_workspace.main.name
+  workspace_resource_id = azurerm_log_analytics_workspace.main[0].id
```

Comment:
What if the count is zero? Will there be any friendly error message / default behaviors, or anything will crash?

Reply:
It will not be zero, it is created in the same condition with log analytics workspace. Two cases have been test in fixture test, enable log analytics or not.